server/ast/expr.go

```diff
@@ -183,12 +185,21 @@ func nodeExpr(node tree.Expr) (vitess.Expr, error) {
 			Else:  else_,
 		}, nil
 	case *tree.CastExpr:
-		//TODO: finish the implementation
+		expr, err := nodeExpr(node.Expr)
```

Comment:
We need to handle `CastExpr.SyntaxMode`, preferably erroring on all non-supported forms (which are probably the non-default ones). For every new node that we're implementing, I'll always check that all public fields are being handled in some way, whether with an error, TODO, or comment explaining why it's ignorable, so we can easily keep track of what we don't yet support.

That's also why I always explicitly list every single enum value is a `switch` statement, and error on the `default` case, as although we handle all of the values _today_, any future yacc additions may change that, and I want the program to properly error. There's nothing worse than a database that seems to work and silently behaves in an unexpected way, as that can lead to unexpected data loss if you e.g. thought you inserted something that never actually went through.

Reply:
Done